### PR TITLE
feat: migrate smb API group to library model

### LINKED ISSUE #237: Refactor the SMB API group to become a library

### PR DESCRIPTION
**Pull request by alexander-ding**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
/kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Part of https://github.com/kubernetes-csi/csi-proxy/issues/217 (migrating one API Group to v2's library based model). The v1 model will stick around until all API groups are migrated.

**Which issue(s) this PR fixes**:

Fixes #237

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
Deferring release note till all API groups are migrated.

### CODE REVIEW

File: integrationtests/smb_test.go [hunk 1]
```diff
@@ -1,138 +1,89 @@
...
+	smbapi "github.com/kubernetes-csi/csi-proxy/pkg/smb/api"
+)
 
-func stringWithCharset(length int, charset string) string {
-	b := make([]byte, length)
-	for i := range b {
-		b[i] = charset[seededRand.Intn(len(charset))]
-	}
-	return string(b)
+func TestSmbAPIGroup(t *testing.T) {
```
Comment (mauriciopoppe) [inline integrationtests/smb_test.go:18]: Change to SMB tracked in https://github.com/kubernetes-csi/csi-proxy/issues/246

File: integrationtests/utils.go [hunk 2]
```diff
@@ -346,3 +346,113 @@ func volumeInit(volumeClient volume.Interface, t *testing.T) (*VirtualHardDisk,
 	}
 	return vhd, volumeID, vhdCleanup
 }
+
+const letterset = "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func stringWithCharset(length int, charset string) string {
```
Comment (mauriciopoppe) [inline integrationtests/utils.go:354]: These are utils not used by any other integration test, I'd limit the scope of these functions to be inside the SMB test

I think `volumeInit` was used in many API Groups so I guess that one is ok

Comment (alexander-ding) [reply integrationtests/utils.go]: moved back in latest commit :)

**Comment by mauriciopoppe** [comment]
Saw this in the last run:
```
--- PASS: TestSmbAPIGroup (62.12s)
    --- PASS: TestSmbAPIGroup/v1alpha1SmbTests (19.81s)
    --- PASS: TestSmbAPIGroup/v1beta1SmbTests (13.25s)
    --- PASS: TestSmbAPIGroup/v1beta2SmbTests (15.24s)
    --- PASS: TestSmbAPIGroup/v1SmbTests (13.82s)
=== RUN   TestSmb
--- PASS: TestSmb (11.88s)
```

/lgtm
/approve

**Pull request merged**